File: front/src/app/core/models/cultivo.ts
import { TipoCultivoResumen } from './tipo-cultivo';

export type EstadoCultivo = 'ACTIVO' | 'FINALIZADO' | 'PERDIDO' | 'EN_ESPERA' | 'LISTO_COSECHA';

export interface Cultivo {
  id: number;
  parcelaId: number;
  parcelaNombre: string;
  tipoCultivo: TipoCultivoResumen;
  nombrePersonalizado?: string;
  fechaSiembra: string;
  fechaCosechaEstimada?: string;
  estado: EstadoCultivo;
  notas?: string;
  activo: boolean;
  createdAt: string;
}

export interface CrearCultivoRequest {
  parcelaId: number;
  tipoCultivoId: number;
  nombrePersonalizado?: string;
  fechaSiembra: string;
  fechaCosechaEstimada?: string;
  notas?: string;
}

export interface ActualizarCultivoRequest {
  nombrePersonalizado?: string;
  fechaCosechaEstimada?: string;
  estado?: EstadoCultivo;
  notas?: string;
}

File: back/src/main/java/com/tfg/agrogestion/domain/cultivo/entity/Cultivo.java
package com.tfg.agrogestion.domain.cultivo.entity;

import com.tfg.agrogestion.domain.parcela.entity.Parcela;
import com.tfg.agrogestion.domain.tipocultivo.entity.TipoCultivo;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "cultivos")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Cultivo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parcela_id", nullable = false)
    private Parcela parcela;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tipo_cultivo_id", nullable = false)
    private TipoCultivo tipoCultivo;

    @Column(name = "nombre_personalizado", length = 150)
    private String nombrePersonalizado;

    @Column(name = "fecha_siembra", nullable = false)
    private LocalDate fechaSiembra;

    @Column(name = "fecha_cosecha_estimada")
    private LocalDate fechaCosechaEstimada;

    @Column(nullable = false, length = 20)
    private String estado;

    @Column(columnDefinition = "NVARCHAR(1000)")
    private String notas;

    @Column(nullable = false)
    private Boolean activo;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @CreatedBy
    @Column(name = "created_by", length = 100)
    private String createdBy;

    @LastModifiedBy
    @Column(name = "updated_by", length = 100)
    private String updatedBy;
}
